### PR TITLE
providers/packet: add gateway attributes

### PR DESCRIPTION
**Pull request by johananl**
## General

This PR adds network gateway information to the attributes on Packet.

## Rationale

Having the gateway address in the attributes file is useful because it allows systemd units which rely on the gateway addresses to read the information via an environment file without having to run shell commands in `ExecStartPre` directives and write error-prone scripting to parse IP addresses from shell output.

### CODE REVIEW

File: src/providers/packet/mock_tests.rs [hunk 1]
```diff
@@ -53,7 +53,44 @@ fn test_packet_attributes() {
...
         "network": {
             "interfaces": [],
-            "addresses": [],
+            "addresses": [
+              {
+                "id": "fde74ec8-bc24-43ca-a852-875bd6e10bee",
+                "address_family": 4,
+                "netmask": "255.255.255.254",
+                "public": true,
+                "management": true,
```
Comment (lucab) [inline src/providers/packet/mock_tests.rs:62]: Out of curiosity, what does this field mean?

File: README.md [hunk 2]
```diff
@@ -101,8 +101,11 @@ On CoreOS Container Linux, the supported providers and metadata are [somewhat di
       - AFTERBURN_PACKET_HOSTNAME
       - AFTERBURN_PACKET_PLAN
       - AFTERBURN_PACKET_IPV4_PUBLIC_0
+      - AFTERBURN_PACKET_IPV4_PUBLIC_GW_0
       - AFTERBURN_PACKET_IPV4_PRIVATE_0
+      - AFTERBURN_PACKET_IPV4_PRIVATE_GW_0
       - AFTERBURN_PACKET_IPV6_PUBLIC_0
+      - AFTERBURN_PACKET_IPV6_PUBLIC_GW_0
```
Comment (lucab) [inline README.md]: It looks like this is missing the private IPv6 address/gateway fields.

File: docs/container-linux-legacy.md [hunk 3]
```diff
@@ -84,8 +84,11 @@ On Container Linux, the supported cloud providers and their respective metadata
       - COREOS_PACKET_HOSTNAME
       - COREOS_PACKET_PLAN
       - COREOS_PACKET_IPV4_PUBLIC_0
+      - COREOS_PACKET_IPV4_PUBLIC_GW_0
       - COREOS_PACKET_IPV4_PRIVATE_0
+      - COREOS_PACKET_IPV4_PRIVATE_GW_0
       - COREOS_PACKET_IPV6_PUBLIC_0
+      - COREOS_PACKET_IPV6_PUBLIC_GW_0
```
Comment (lucab) [inline docs/container-linux-legacy.md]: Same comment as the main README one.

File: README.md [hunk 4]
```diff
@@ -101,8 +101,11 @@ On CoreOS Container Linux, the supported providers and metadata are [somewhat di
       - AFTERBURN_PACKET_HOSTNAME
       - AFTERBURN_PACKET_PLAN
       - AFTERBURN_PACKET_IPV4_PUBLIC_0
+      - AFTERBURN_PACKET_IPV4_PUBLIC_GW_0
```
Comment (lucab) [inline README.md]: Here and everywhere, I'd probably spell out `GATEWAY` entirely.

**Comment by lucab** [comment]
Thanks, great PR! I only have few minor nits on it.

File: src/providers/packet/mock_tests.rs (hunk 1 above)
Comment (johananl) [reply src/providers/packet/mock_tests.rs:62]: AFAIK a "management IP" is an IP address that is bound to a Packet device (server). Packet offers also "Elastic IPs", customer-provided IPs etc.
Here, for example, is what what an Elastic IP (which is a non-management IP) looks like when attached to a device:

```json
{
  "id": "d392685e-072c-419b-a588-402b48d77bdc",
  "address_family": 4,
  "netmask": "255.255.255.255",
  "created_at": "2020-02-18T09:27:51Z",
  "public": true, 
  "cidr": 32,
  "management": false,
  "enabled": true, 
  "network": "147.1.1.1",
  "address": "147.1.1.1",
  "gateway": "147.1.1.1",
  "parent_block": {
    "network": "147.1.1.1",
    "netmask": "255.255.255.255",
    "cidr": 32,
    "href": "/ips/fdaa9c07-3dbc-47fd-8970-c3161888136b"
  }
}
```

File: README.md (hunk 4 above)
Comment (johananl) [reply README.md]: Done.

**Review by lucab — APPROVED** [review]
LGTM!

**Pull request merged**